apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -261,6 +265,47 @@ impl LightSpanData {
     }
 }
 
+/// An externally updateable gauge for "apollo_router_span_lru_size".
+///
+/// When observed, it reports the most recently stored value (give or take atomicity looseness).
+///
+/// This *could* be generalised to any kind of gauge, but we should ideally have gauges that can just
+/// observe their accurate value whenever requested. The externally updateable approach is kind of
+/// a hack that happens to work here because we only have one place where the value can change, and
+/// otherwise we might have to use an inconvenient Mutex or RwLock around the entire LRU cache.
+#[derive(Debug)]
+struct SpanLruSizeInstrument {
+    value: Arc<AtomicU64>,
+    _gauge: ObservableGauge<u64>,
+}
```

Comment:
As mentioned in the doc comment, this is a hack that kind of mimicks what `tracing::info!(value.)` was doing (storing the value whenever it is updated, and reading the stored value when requested).

I'm not sure about this metric, should we keep it at all in 2.0?

Reply:
I think we should keep it. It's really helpful to debug and could potentially find the root cause of a leak or so.